### PR TITLE
Update Docstrings in all classes related to RandomVariables

### LINKED ISSUE #146: Write a warning into RandomVariable about order of operations
Samples from random variables with a fixed seed are not stable with respect to the order of operations (e.g. slicing, masking). Write a warning into the `probnum.prob.RandomVariable` docs about this, including an example with `Normal`.

### LINKED ISSUE #199: Documentation of changes in `RandomVariable`
* [ ] We should define what we mean by the "distribution of a random variable"
* [ ] We should make the analogy to numpy arrays explicit in the documentation
* [ ] We should somehow document that we only want to use `np.generic` and `np.ndarray` internally, if possible
* [ ] Maybe we should quickly describe the steps to add a new `RandomVariable` subclass, including how to define new arithmetic operations?

Leftovers from the PR #164:
- [ ] Update the quickstart guide and notebooks explaining the use of random variables
- [ ] Update the documentation thoroughly in all changed files

### PR DESCRIPTION
**Pull request by JonathanWenger**
This PR updates and adds missing docstrings to the recently refactored classes associated with `RandomVariable`.

**Docstrings**
- [x] Update the documentation thoroughly in all changed files
- [x] define what we mean by the "distribution of a random variable"
- [x] Warning about order of operations in `RandomVariable` (closes #146)
- [x] make the analogy to numpy arrays explicit in the documentation

**Tutorials and Quickstart Guide**:
- [x] Update the quickstart guide and notebooks explaining the use of random variables

**Developer Guides**:
- ~[ ] describe the steps to add a new `RandomVariable` subclass, including how to define new arithmetic operations~

**Other:**
- [x] alter the `__repr__` of RandomVariables (see also https://github.com/probabilistic-numerics/probnum/issues/199#issuecomment-683436045)
- [x] document that we only want to use np.generic and np.ndarray internally, if possible

This PR closes #199.

_Note: You can view the built documentation by going to the "Details" page of the `docs/readthedocs.org:probnum` check, which builds a version of the documentation corresponding to this PR._

### CODE REVIEW

File: src/probnum/random_variables/_random_variable.py [hunk 1]
```diff
@@ -62,26 +47,76 @@ class RandomVariable(Generic[_ValueType]):
...
+        :class:`~numpy.random.RandomState` instance.
+    parameters :
+        Parameters of the distribution of the random variable.
+    sample :
+        Callable implementing sampling from the random variable.
+    in_support :
+        Callable checking whether the random variable takes value ``x`` with non-zero
+        probability, i.e. if ``x`` is in the support of its distribution.
+    cdf :
+        The cumulative distribution function of the random variable.
```
Comment (marvinpfoertner) [inline src/probnum/random_variables/_random_variable.py:62]: Should we add the "Callable [...]" to the remaining arguments?

**Review by marvinpfoertner — COMMENTED** [review]
I added a comment every time something didn't read smoothly, so my comments are relatively noisy. Not everything needs fixing.

File: src/probnum/random_variables/_constant.py [hunk 2]
```diff
@@ -17,33 +19,44 @@
 _ValueType = TypeVar("ValueType")
 
 
-class Dirac(_random_variable.DiscreteRandomVariable[_ValueType]):
+class Constant(_random_variable.DiscreteRandomVariable[_ValueType]):
```
[the remark this answers is not in the document]
Comment (JonathanWenger) [reply src/probnum/random_variables/_constant.py:22]: Dirac is a remnant from when we had `Distribution` classes. There is no such thing as a Dirac random variable. Only a Dirac measure or a Dirac distribution (in the generalized function sense). However a (almost surely) constant random variable is an established concept.

File: docs/source/public_api/diffeq.rst [hunk 3]
```diff
@@ -1,5 +1,10 @@
 probnum.diffeq
 ==============
 
+Differential Equations.
+
+This package defines common dynamical models and probabilistic solvers for differential
+equations.
+
```
[the remark this answers is not in the document]
Comment (JonathanWenger) [reply docs/source/public_api/diffeq.rst:8]: @nathanaelbosch @pnkraemer feel free to take a look at the doc changes I made to this file, but they are pretty minor.

File: src/probnum/filtsmooth/__init__.py [hunk 4]
```diff
@@ -1,7 +1,3 @@
-"""
-Bayesian filtering and smoothing.
-"""
-
```
[the remark this answers is not in the document]
Comment (JonathanWenger) [reply src/probnum/filtsmooth/__init__.py:4]: I had moved them to the .rst files to be able to include images in the doc pages for subpackages. Keeping them around made them show up twice. This is now solved with the `automodapi` setting `:no-main-docstr:` and I readded the module level docstrings. This now also does not generate `pylint` errors anymore.

File: src/probnum/random_variables/_normal.py [hunk 5]
```diff
@@ -27,50 +27,43 @@
...
 
-    The Gaussian distribution is ubiquitous in probability theory, since
-    it is the final and stable or equilibrium distribution to which
-    other distributions gravitate under a wide variety of smooth
-    operations, e.g., convolutions and stochastic transformations.
-    One example of this is the central limit theorem. The Gaussian
-    distribution is also attractive from a numerical point of view as it
-    is maintained through many transformations (e.g. it is stable).
+    Gaussian random variables are ubiquitous in probability theory, since the
+    Gaussian is the final and stable or equilibrium distribution to which other
```
[the remark this answers is not in the document]
Comment (JonathanWenger) [reply src/probnum/random_variables/_normal.py:38]: This is synonymous with equilibrium in this context. I removed it.

**Review by marvinpfoertner — APPROVED** [review]

**Pull request merged**